### PR TITLE
If channel_max is 0, use server's channel_max

### LINKED ISSUE #242: channel_max negotiation
Be able to negotiate (use the lowest, but not zero, of the suggested value from the server or the client) connection tune parameters.

### PR DESCRIPTION
**Pull request by carlhoerberg**
Fixes #242

### CODE REVIEW

File: librabbitmq/amqp_socket.c [hunk 1]
```diff
@@ -1354,7 +1354,7 @@ static amqp_rpc_reply_t amqp_login_inner(amqp_connection_state_t state,
     server_heartbeat = s->heartbeat;
   }
 
-  if (server_channel_max != 0 && server_channel_max < channel_max) {
+  if (server_channel_max != 0 && (server_channel_max < channel_max || channel_max == 0)) {
```
Comment (alanxz) [inline librabbitmq/amqp_socket.c]: Please reformat this so that it fits in 80 columns

**Comment by alanxz** [comment]
Could you add a comment in the [`amqp_login()`](https://github.com/alanxz/rabbitmq-c/blob/ceca34823beb017e9615e77bd03fb3f54d64f50b/librabbitmq/amqp.h#L1749) and [`amqp_login_with_properties()`](https://github.com/alanxz/rabbitmq-c/blob/ceca34823beb017e9615e77bd03fb3f54d64f50b/librabbitmq/amqp.h#L1797) documentation describing this behavior?

LGTM otherwise, thanks for the contribution.

**Comment by carlhoerberg** [comment]
Thank you for the feedback. Do you want me to rebase to make the commit log nicer or is this ok?

**Comment by alanxz** [comment]
Could you rebase this down to 1 commit?

Thanks

**Comment by carlhoerberg** [comment]
Done

**Comment by alanxz** [comment]
Merged.  Thanks for the contribution!

**Pull request merged**